controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -488,6 +489,44 @@ func (w *Workload) ClusterStatus(ctx context.Context) (ClusterStatus, error) {
 	return status, nil
 }
 
+// ForwardEtcdLeadership forwards etcd leadership to the first follower
+func (w *Workload) ForwardEtcdLeadership(ctx context.Context, machine *clusterv1.Machine) error {
+	if machine == nil || machine.Status.NodeRef == nil {
+		// Nothing to do, no node for Machine
+		return nil
+	}
+
+	etcdClient, err := w.etcdClientGenerator.forNode(ctx, machine.Status.NodeRef.Name)
+	if err != nil {
+		return errors.Wrap(err, "failed to create etcd Client")
+	}
+
+	// List etcd members. This checks that the member is healthy, because the request goes through consensus.
+	members, err := etcdClient.Members(ctx)
+	if err != nil {
+		return errors.Wrap(err, "failed to list etcd members using etcd client")
+	}
+
+	currentMember := etcdutil.MemberForName(members, machine.Status.NodeRef.Name)
+
+	if currentMember.ID != etcdClient.LeaderID {
+		return nil
+	}
+
+	// If current member is leader, move leadship to first follower
+	for _, member := range members {
+		if member.ID != currentMember.ID {
+			err := etcdClient.MoveLeader(ctx, member.ID)
```

Comment:
This should make sure that the new member isn't a machine that's going to be deleted, a (potential) simple solution would be to always pick the last created machine/etcd member

Reply:
@alexander-demichev do you have time to tackle the above? or we can do it in a follow-up PR

Reply:
If you are fine with this PR then feel free to merge. A follow-up sounds good, if that's not urgent I can try to make something during the week.